types/src/transaction/transaction_entry_point.rs

```diff
@@ -198,7 +228,10 @@ impl TransactionEntryPoint {
             | TransactionEntryPoint::Undelegate
             | TransactionEntryPoint::Redelegate
             | TransactionEntryPoint::ActivateBid
-            | TransactionEntryPoint::ChangeBidPublicKey => false,
+            | TransactionEntryPoint::ChangeBidPublicKey
+            // TODO(jck): make sure this is correct
```

Comment:
Holds epoch is necessary to calculate available balance, so any operation that includes a balance check requires it.

As I understand it, reservations don't involve a balance check and thus Add / Cancel Reservation should not require holds epoch.